test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_cad.cxx

```diff
@@ -365,6 +370,238 @@ TEST (t8_gtest_geometry_cad, jacobian)
+
+#if T8_WITH_OCC
+/* The test checks if the mapping algorithms for curved 2d elements shift values on a curved cad curve correctly. */
+class class_2d_element_curved_cad_curve: public testing::TestWithParam<t8_eclass_t> {
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Curve cad_curve;
+    TColgp_Array1OfPnt point_array (1, 3);
+
+    /*  x--> u-parameter
+    *   
+    *  ----____       curve       ____----
+    *           ----_________----
+    * 
+    *  0               0.5              1
+    */
+
+    point_array (1) = gp_Pnt (test_ref_coords_out[0], test_ref_coords_out[1], test_ref_coords_out[2]);
+    point_array (2) = gp_Pnt (test_ref_coords_out[3], test_ref_coords_out[4], test_ref_coords_out[5]);
+    point_array (3) = gp_Pnt (test_ref_coords_out[6], test_ref_coords_out[7], test_ref_coords_out[8]);
+
+    cad_curve = GeomAPI_PointsToBSpline (point_array).Curve ();
+    shape = BRepBuilderAPI_MakeEdge (cad_curve).Edge ();
+
+    num_vertices = t8_eclass_num_vertices[eclass];
+  }
+
+  void
+  TearDown () override
+  {
+    /* The cmesh is destroyed in the test itself. */
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  TopoDS_Shape shape;
+  /* Saving the corner vertices for the given element class. */
+  size_t num_vertices;
+  const double vertices_tri[27] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0,
+                                    0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0 };
+  const double vertices_quad[48] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0,
+                                     1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0,
+                                     0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0 };
+  /* Edges are parametrized in one parameter u. The array contains the parameters
+   * each vertex of the edge has on the linked curve. */
+  double params_tri[6] = { 0, 1, 1, 0, 0, 1 };
+  double params_quad[8] = { 0, 1, 1, 0, 1, 0, 0, 1 };
+  /* The array prescribes the linkage of the element. No face is linked. */
+  std::array<int, 1> faces = { 0 };
+  std::array<int, 8> edges = { 0 };
+
+  const int linked_edge_tri[3] = { 2, 1, 0 };
+  const int linked_edge_quad[4] = { 2, 0, 3, 1 };
+
+  const double test_ref_coords_tri_in[27] = { 0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.5, 0.5,
+                                              0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.5, 0.0, 1.0, 1.0, 0.0 };
+  const double test_ref_coords_quad_in[36]
+    = { 0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.5, 0.0, 0.0, 0.0, 0.0,
+        1.0, 1.0, 0.0, 0.5, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.5, 0.0, 1.0, 1.0, 0.0 };
+
+  const double test_ref_coords_out[9] = { 0.0, 0.0, 0.0, 0.5, -0.2, 0.0, 1.0, 0.0, 0.0 };
```

Comment:
This is a lot of code duplication. Can we use the same fixture for both tests? We can then define the variables which are different in the functions itself, or we can derive subclasses for the different tests. Or we use scope limited global variables (least preferred option).